src/hotspot/share/runtime/thread.hpp

```diff
@@ -1247,8 +1249,23 @@ class JavaThread: public Thread {
     return (_suspend_flags & _thread_suspended) != 0;
   }
 
-  bool is_VTMT_disabler() const                  { return _is_VTMT_disabler; }
-  bool is_in_VTMT() const                        { return _is_in_VTMT; }
+  bool is_VTMT_disabler() const {
+#if INCLUDE_JVMTI
+    return _is_VTMT_disabler;
+#else
+    fatal("Should only be called with JVMTI enabled");
+    return false;
+#endif
+  }
+
+  bool is_in_VTMT() const {
+#if INCLUDE_JVMTI
+    return _is_in_VTMT;
+#else
+    fatal("Should only be called with JVMTI enabled");
+    return false;
+#endif
+  }
```

Comment:
If you #ifdef out the API, then you know it's not being called, thus no need for #else sections and calls to `fatal()`.

Reply:
All right, that is indeed cleaner. Please see new commit.